mackerel-plugin-docker/docker.go

```diff
@@ -123,6 +124,61 @@ func (m DockerPlugin) findPrefixPath() (string, error) {
 	return "", errors.New("No prefix path is found.")
 }
 
+type pathBuilder struct {
+	prefix   string
+	pathType pathType
+}
+
+type pathType uint8
+
+const (
+	pathUnknown pathType = iota
+	pathDocker
+	pathLxc
+	pathSlice
+)
+
+func newPathBuilder() (*pathBuilder, error) {
+	prefixPath, err := findPrefixPath()
+	if err != nil {
+		return nil, err
+	}
+	pathT, err := guessPathType(prefixPath)
+	if err != nil {
+		return nil, err
+	}
+	return &pathBuilder{
+		prefix:   prefixPath,
+		pathType: pathT,
+	}, nil
+}
+
+func (pb *pathBuilder) build(id, metric, postfix string) string {
+	switch pb.pathType {
+	case pathDocker:
+		return fmt.Sprintf("%s/%s/docker/%s/%s.%s", pb.prefix, metric, id, metric, postfix)
+	case pathLxc:
+		return fmt.Sprintf("%s/%s/lxc/%s/%s.%s", pb.prefix, metric, id, metric, postfix)
+	case pathSlice:
+		return fmt.Sprintf("%s/%s/system.slice/docker-%s.scope/%s.%s", pb.prefix, metric, id, metric, postfix)
+	default:
+		return ""
+	}
+}
+
+func guessPathType(prefix string) (pathType, error) {
+	if ok, err := exists(prefix + "/memory/system.slice/"); ok && err == nil {
+		return pathSlice, nil
+	}
+	if ok, err := exists(prefix + "/memory/docker/"); ok && err == nil {
+		return pathDocker, nil
+	}
+	if ok, err := exists(prefix + "/memory/lxc/"); ok && err == nil {
+		return pathLxc, nil
+	}
+	return pathUnknown, fmt.Errorf("can't resolve runtime metrics path")
```

Comment:
[nit] It may be better to show a possibility that this error may occur in the case of no container exists.